src/test/java/com/epam/jamp2/service/impl/CalculationCommandExecutionServiceTest.java

```diff
@@ -0,0 +1,158 @@
+package com.epam.jamp2.service.impl;
+
+import com.epam.jamp2.model.*;
+import com.epam.jamp2.rest.FixerioServiceProxy;
+import okhttp3.Request;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.math.BigDecimal;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CalculationCommandExecutionServiceTest {
+
+    @Mock
+    private FixerioServiceProxy fixerioServiceProxy;
+
+    @InjectMocks
+    private FxRatesServiceImpl fxRatesServiceImpl;
+
+    private CalculationCommandExecutionServiceImpl calculationCommandExecutionService;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    /**
+     * Read rates from rates.ser and mock remote api
+     *
+     * @throws Exception reading rates data from disk failed exception
+     */
+    @Before
+    public void setupMocks() throws Exception {
+        //Sample currency converting rates from rates.ser
+        FileInputStream fis = new FileInputStream("src/test/resources/rates.ser");
+        ObjectInputStream ois = new ObjectInputStream(fis);
+        Map<String, Map<String, BigDecimal>> testRates = (Map<String, Map<String, BigDecimal>>) ois.readObject();
+        ois.close();
+
+        //Mock the getRates method of fixerioServiceProxy
+        when(fixerioServiceProxy.getRates(any(String.class))).thenAnswer(invocationOnMock ->
+                new Call<FixerioResponse>() {
+                    @Override
+                    public Response<FixerioResponse> execute() throws IOException {
+                        FixerioResponse response = new FixerioResponse();
+                        String currency = (String) invocationOnMock.getArguments()[0];
+                        response.setRates(testRates.get(currency.toUpperCase()));
+                        return Response.success(response);
+                    }
+
+                    @Override
```

Comment:
I think no need to override those methods you don't use, it will make your code more readable.